docs/source/format/CanonicalExtensions.rst

```diff
@@ -283,6 +283,77 @@ UUID
    A specific UUID version is not required or guaranteed. This extension represents
    UUIDs as FixedSizeBinary(16) with big-endian notation and does not interpret the bytes in any way.
 
+Other
+=====
+
+Other represents a type or array that one Arrow-based system received from an
+external (likely non-Arrow) system, but cannot interpret itself.  In this
+case, the Other type explicitly communicates the name and presence of a field
+to downstream clients.
+
+For example:
+
+* A Flight SQL service may support connecting external databases.  In this
+  case, its catalog (``GetTables`` etc.) should reflect the names and types of
+  tables in external databases.  But those external systems may support types
+  it does not recognize.  Instead of erroring or silently dropping columns
+  from the catalog, it can use the Other[Null] type to report that a column
+  exists with a particular name and type name in the external database; this
+  lets clients know that a column exists, but is not supported.
+
+* The ADBC PostgreSQL driver, because of how the PostgreSQL wire protocol
+  works, may get bytes for a field whose type it does not recognize (say, a
+  geospatial type).  It can still return the bytes to the application which
+  may be able to parse the data itself.  In that case, it can use the
+  Other[binary] type to return the column data.  The Other type differentiates
+  the column from actual binary columns.
+
+Of course, the intermediate system *could* implement a custom extension type
+for these example types.  But there is no way in general that every type can
+be known in advance.  In such cases, the Other type allows the system to
+explicitly note that it does not support some type or field, without silently
+losing data or sending irrelevant errors.  It could also pretend to support
+the types by making up extension types on the fly.  But this misleads
+downstream systems who cannot tell if the type is supported or not.
```

Comment:
@lidavidm thanks for the expanded explanation here, I think that makes the scope much clearer now